### DISCUSSION TITLE
How to use Date/Time/Datetime in model?

### DISCUSSION THREAD

**Discussion by qarmin** [rwf2/Rocket#2273] [Questions]
In code I use such function
```
#[openapi(tag = "Patient")]
#[post("/patients", data = "<new_patient>")]
pub async fn create_patient(
    db: &State<DatabaseConnection>,
    new_patient: Json<NewPatientInfo>,
) -> Result<Json<PatientInfo>, ErrorResponder> {}
```
which require to implements following traits
```
#[derive(Serialize, Deserialize, JsonSchema)]
```
so when I added one field with Date type,
```
    pub(crate) birth_date: Date,
```
 I have following errors
![Screenshot from 2022-07-17 09-41-49](https://user-images.githubusercontent.com/41945903/179388891-69c197aa-5f50-48ea-95bf-f9a305cda43c.png)
Is there any macro/method that would implement such traits automatically?

**Comment by SergioBenitez** [comment]
The `time` crate does support serde, but you need to enable it's support directly. You can do this by adding the time crate to your Cargo.toml:

```toml
time = { version = "0.3", features = ["serde"] }
```

You can continue to use the `rocket::time` reexport. 

The `jsonschema` crate is entirely external to Rocket, so you'll need to consult their documentation and see if/how they support `time`.

**Comment by qarmin** [comment]
Looks that this is a problem of Schemars library which is used by OpenAPI swagger generator - https://github.com/GREsau/schemars/issues/104
